<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>NGS - software notes</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "NGS";
    var mkdocs_page_input_path = "NGS.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> software notes</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="" href="../index.md">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../AA/">Association</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../AI/">AI</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../CRISPR/">CRISPR</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">NGS</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#review">Review</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#ngs-pipeline">NGS pipeline</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#annotation">Annotation</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#case-studies">Case studies</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#agotron-detection">Agotron detection</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#alignment-and-variant-calling-tutorial">Alignment and variant calling tutorial</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#bowtie-scaling">bowtie-scaling</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#exome-sequencing-analysis">Exome sequencing analysis</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#cnv-detection">CNV detection</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#snp-discovery">SNP discovery</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#tss">TSS</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#comparison-of-gene-expression-pipelines-on-rna-seq-sequencing-data">Comparison of gene expression pipelines on RNA-seq sequencing data.</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#gsnap-mapsplice-rum-star-rna-seq-pipeline">GSNAP, MapSplice, RUM, STAR, RNA-seq pipeline</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#mendelian-rna-seq">Mendelian RNA-seq</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#sra-toolkit-tophat">sra-toolkit, tophat</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../SL/">SL</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../misc/">misc</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../pharmacogenomics/">Pharmacogenomics</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../scRNASeq/">scRNASeq</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../prottrans/">prottrans</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">software notes</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>NGS</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="benchmarks-and-pipelines">Benchmarks and pipelines</h1>
<h2 id="review">Review</h2>
<p>Pabinger S, et al. (2014). A survey of tools for variant analysis of next-generation genome sequencing data. <em>Brief Bioinformatics</em> 15(2):256-278.</p>
<h2 id="ngs-pipeline">NGS pipeline</h2>
<p>https://www.hgsc.bcm.edu/software/mercury</p>
<h2 id="annotation">Annotation</h2>
<p>https://sites.google.com/site/jpopgen/wgsa</p>
<h2 id="case-studies">Case studies</h2>
<p>Lagana A, et al. (2018). Precision medicine for relapsed multiple myeloma on the basis of an integrative multiomics approach. <em>JCO Prec Oncol.</em> Data Suppl, 
http://ascopubs.org/doi/suppl/10.1200/PO.18.00019</p>
<p>Lu X-M, et al. (2018). Association of breast and ovarian cancers with predisposition genes identified by large-scale sequencing. <em>JAMA Oncol</em>, doi:10.1001/jamaoncol.2018.2956.</p>
<p>Mestek-Boukhibar L, et al. (2018). Rapid Paediatric Sequencing (RaPS): comprehensive real-life workflow for rapid diagnosis of critically ill children. <em>J Med Genet</em>, doi:10.1136/jmedgenet-2018-105396</p>
<p>Castel SE, et al. (2018). Modified penetrance of coding variants by cis-regulatory variation contributes to disease risk.<em>Nat Genet</em>, https://doi.org/10.1038/s41588-018-0192-y.</p>
<p>Dixon JR, et al. (2018). Integrative detection and analysis of structural variation in cancer genomes. <em>Nat Genet</em>, https://www.nature.com/articles/s41588-018-0195-8</p>
<p>Wood DE, et al. (2018). A machine learning approach for somatic mutation discovery. Sci. Transl. Med. 10, eaar7939 (2018) DOI: 10.1126/scitranslmed.aar7939</p>
<h2 id="agotron-detection">Agotron detection</h2>
<p>The following is according to https://github.com/ncrnalab/agotron_detector as described in</p>
<blockquote>
<p>Hansen TB (2018). Detecting Agotrons in Ago CLIPseq Data. in Vang Ã˜rom UA (ed) miRNA Biogenesis-Methods and Protocols, Chapter 17, 221-232. Springer.</p>
</blockquote>
<pre><code class="language-bash">wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz
tar -zxvf chromFa.tar.gz
cat *.fa &gt; hg19.fa
samtools faidx hg19.fa
bowtie2-build hg19.fa hg19
# GSE78059
for srr in 008/SRR3177718/SRR3177718 009/SRR3177719/SRR3177719 000/SRR3177720/SRR3177720 001/SRR3177721/SRR3177721 002/SRR3177722/SRR3177722 003/SRR3177723/SRR3177723
do
   wget ftp.sra.ebi.ac.uk/vol1/fastq/SRR317/$srr.fastq.gz
done
trim_galore -A TCAGTCACTTCCAGC -length 18 *.fastq.gz
for i in *_trimmed.fq.gz
do
    echo $i
    bowtie2 -q --local -x hg19 -U $i | samtools sort - &gt; $i.sort.bam    
    samtools index $i.sort.bam

done
python UCSC_intron_retriever.py | python analyzer.py -g hg19.fa | Rscript annotater.R
</code></pre>
<p>Note that it is easier to implement with <code>prefetch</code> as shown below.</p>
<h2 id="alignment-and-variant-calling-tutorial">Alignment and variant calling tutorial</h2>
<p>See https://github.com/ekg/alignment-and-variant-calling-tutorial. Note that E.coli_K12_MG1655.fa is unavailable any more, instead we have to download it directly from NCBI, 
https://www.ncbi.nlm.nih.gov/nuccore/556503834, choose FASTA (text), to reach https://www.ncbi.nlm.nih.gov/nuccore/NC_000913.3?report=fasta&amp;log$=seqview&amp;format=text and save
to a local file, whose empty lines have to be removed, see them with awk '(length($NF)==0){print NR}' E.coli_K12_MG1655.fa.</p>
<p>The fastq-dump generates .fa files, which need to be compressed with gzip.</p>
<h2 id="bowtie-scaling">bowtie-scaling</h2>
<p>https://github.com/BenLangmead/bowtie-scaling</p>
<h2 id="exome-sequencing-analysis">Exome sequencing analysis</h2>
<blockquote>
<p>IMSGC (2018). Low-frequency and rare-coding variation contributes to multiple sclerosis risk. <em>Cell</em>. DOI:https://doi.org/10.1016/j.cell.2018.09.049</p>
</blockquote>
<p>has associate software, https://github.com/cotsapaslab/IMSGCexomechip.</p>
<h2 id="cnv-detection">CNV detection</h2>
<p>CN-Learn is a framework to integrate Copy Number Variant (CNV) predictions made by multiple algorithms using exome sequencing datasets.</p>
<p>https://github.com/girirajanlab/CN_Learn</p>
<p>Pounraja VK, et al. (2019) A machine-learning approach for accurate detection of copy-number variants from exome sequencing. <em>Genome Res</em>.</p>
<h2 id="snp-discovery">SNP discovery</h2>
<p>The following reference discribes several pipelines for SNP discovery.</p>
<blockquote>
<p>Morin PA, Foote AD, Hill CM, Simon-Bouhet B, Lang AR, Louis M (2018). SNP Discovery from Single and Multiplex Genome Assemblies of Non-model Organisms, in Head SR, et al. (eds.), Next Generation Sequencing: Methods and Protocols, Chapter 9, 113-144, Springer.</p>
</blockquote>
<p>whose scripts are available from https://github.com/PAMorin/SNPdiscovery/.</p>
<p>See also https://github.com/sanger-pathogens/snp-sites and the following references,</p>
<blockquote>
<p>Martin J, Schackwitz W, Lipzen A (2018). Genomic Sequence Variation Analysis by Resequencing, in de Vries RP, Tsang A, Grigoriev IV (ed) Fungal Genomics-Methods and Protocols, 2e, Chapter 18, 229-239, Springer.</p>
<p>Raghavachari N, Garcia-Reyero N (eds.) (2018), Gene Expression Analysis-Methods and Protocols, Springer.</p>
</blockquote>
<h2 id="tss">TSS</h2>
<p>Mejia-Guerra MK, et al. (2018). Genome-Wide TSS Identification in Maize. Chapter 14, 239-256, in Yamaguchi N (ed.), Plant Transcription Factors-Methods and Protocols, Springer</p>
<h2 id="comparison-of-gene-expression-pipelines-on-rna-seq-sequencing-data">Comparison of gene expression pipelines on RNA-seq sequencing data.</h2>
<p>http://statapps.ugent.be/tools/AppDGE/</p>
<h2 id="gsnap-mapsplice-rum-star-rna-seq-pipeline">GSNAP, MapSplice, RUM, STAR, RNA-seq pipeline</h2>
<pre><code class="language-bash"># gsnap
wget http://research-pub.gene.com/gmap/src/gmap-gsnap-2018-07-04.tar.gz
tar xfz gmap-gsnap-2018-07-04.tar.gz
cd gmap-2018-07-04
./configure
make
sudo make install
# mapsplice, the latest version from http://protocols.netlab.uky.edu/~zeng/MapSplice-v2.2.1.zip has compiling issue
sudo `which conda` install mapsplice
mapsplice.py
# rum
git clone https://github.com/itmat/rum
cd rum
perl Makefile.PL
make
sudo make install
# STAR
git clone https://github.com/alexdobin/STAR
cd STAR/source
make
</code></pre>
<p>See https://github.com/sanger-pathogens/Bio-RNASeq for RNA-seq pipeline.</p>
<h2 id="mendelian-rna-seq">Mendelian RNA-seq</h2>
<p>https://github.com/komalsrathi/MendelianRNA-seq</p>
<p>The relevant installations:</p>
<pre><code>conda create --name mendelian-rnaseq-env
source activate mendelian-rnaseq-env
conda install -c bioconda snakemake
conda install -c bioconda rna-seqc
conda install -c bioconda gatk
conda install -c biobuilds plink
conda install -c bioconda star
conda install -c bioconda picard
conda install -c bioconda bwa
conda install -c anaconda colorama
conda install -c bioconda misopy
</code></pre>
<h2 id="sra-toolkit-tophat">sra-toolkit, tophat</h2>
<p>These are very straightforward, e.g.,</p>
<pre><code class="language-bash">prefetch -v SRR3534842
fastq-dump --split-files --gzip SRR3534842
</code></pre>
<p>the SRR3534842.sra from prefetch is actually at $HOME/ncbi/public/sra which is split into <code>SRR3534842_1.fastq.gz</code>, <code>SRR3534842_2.fastq.gz</code> at the current directory. See https://www.biostars.org/p/111040/. However, the location may not desirable since it may create a huge .vdi files with VirtualBox -- to get around we do this</p>
<pre><code class="language-bash">cd $HOME
mkdir -p /home/jhz22/D/work/ncbi/public/sra
ln -sf /home/jhz22/D/work/ncbi
</code></pre>
<p>where D is actually a shared folder at Windows.</p>
<p>To run <code>tophat</code>, see https://ccb.jhu.edu/software/tophat/tutorial.shtml</p>
<pre><code class="language-bash">wget https://ccb.jhu.edu/software/tophat/downloads/test_data.tar.gz
tar xvfz test_data.tar.gz
cd test_data
tophat -r 20 test_ref reads_1.fq reads_2.fq
</code></pre>
<h1 id="software">Software</h1>
<h2 id="advntr">adVNTR</h2>
<p>It is a tool for genotyping Variable Number Tandem Repeats (VNTR) from sequence data, https://github.com/mehrdadbakhtiari/adVNTR.</p>
<h2 id="angsd">ANGSD</h2>
<p>ANGSD is a software for analyzing next generation sequencing data, http://www.popgen.dk/angsd/index.php/ANGSD. It is relatively straightforward with GitHub; after</p>
<pre><code class="language-bash">git clone https://github.com/ANGSD/angsd
cd angsd
make
</code></pre>
<p>but the following change is needed on line 468 of <code>misc/msHOT2glf.c</code>: <code>tmppch</code> as in (tmppch=='\0') should be <code>*tmppch</code> as in <code>(*tmppch==''0')</code>, suggested by the compiler.</p>
<h2 id="ubuntu-archive-bamtools-bcftools-bedops-bedtools-blast-ncbi-blast-bowtie2-fastqc-fastx-toolkit-freebayes-hmmer-hisat2-picard-tools-rsem-sambamba-samtools-seqtk-sra-toolkit-subread-tophat-trinityrnaseq-vcftool-vowpal-wabbit">Ubuntu archive: bamtools, bcftools, bedops, bedtools, blast (ncbi-blast+), bowtie2, fastqc, fastx-toolkit, freebayes, hmmer, hisat2, picard-tools, rsem, sambamba, samtools, seqtk, sra-toolkit, subread, tophat, trinityrnaseq, vcftool, vowpal-wabbit</h2>
<p>Install with <code>sudo apt install</code>.</p>
<p>See also https://github.com/lh3/seqtk.</p>
<p>Besides notes above, this is also possible:</p>
<h2 id="bcftools">bcftools</h2>
<pre><code class="language-bash">wget https://github.com/samtools/bcftools/releases/download/1.9/bcftools-1.9.tar.bz2
tar jfx bcftools-1.9.tar.bz2
cd bcftools-1.9
./configure --prefix=$HPC_WORK
make
make install
</code></pre>
<p>It is necessary to set the environment variables to enable plugins, so we could generate a version at $HPC_WORK/bin instead,</p>
<pre><code class="language-bash">#!/usr/bin/bash

export BCFTOOLS_PLUGINS=$HPC_WORK/bcftools-1.9/plugins
$HPC_WORK/bcftools-1.9/bcftools &quot;$@&quot;
</code></pre>
<p>We invoke <code>bcftools +check_ploidy my.vcf.gz</code> Interestingly, this also save space!</p>
<h2 id="bowtie2">bowtie2</h2>
<p>The project home is https://sourceforge.net/projects/bowtie-bio, whereby</p>
<pre><code class="language-bash">wget https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.3.4.1/bowtie2-2.3.4.1-linux-x86_64.zip
wget https://sourceforge.net/projects/bowtie-bio/files/bowtie2/2.3.4.1/bowtie2-2.3.4.1-source.zip
unzip bowtie2-2.3.4.1-linux-x86_64.zip
cd bowtie2-2.3.4.1-linux-x86_64/
</code></pre>
<p>The test is then self-contained,</p>
<pre><code class="language-bash">export BT2_HOME=/home/jhz22/D/genetics/bowtie2-2.3.4.1-linux-x86_64

$BT2_HOME/bowtie2-build $BT2_HOME/example/reference/lambda_virus.fa lambda_virus
$BT2_HOME/bowtie2 -x lambda_virus -U $BT2_HOME/example/reads/reads_1.fq -S eg1.sam
$BT2_HOME/bowtie2 -x $BT2_HOME/example/index/lambda_virus -1 $BT2_HOME/example/reads/reads_1.fq -2 $BT2_HOME/example/reads/reads_2.fq -S eg2.sam

samtools view -bS eg2.sam &gt; eg2.bam
samtools sort eg2.bam -o eg2.sorted.bam
samtools mpileup -uf $BT2_HOME/example/reference/lambda_virus.fa eg2.sorted.bam | bcftools view -Ov - &gt; eg2.raw.bcf
bcftools view eg2.raw.bcf
</code></pre>
<p>Like samtools, etc. it is possible to involve <code>sudo apt install bowtie2</code>.</p>
<h2 id="cnvkit">CNVkit</h2>
<p>Genome-wide copy number from high-throughput sequencing, available from
https://cnvkit.readthedocs.io/en/stable/</p>
<h2 id="cutadapt-trimgalore">cutadapt, TrimGalore</h2>
<p>A prerequesite is to install cython.</p>
<pre><code class="language-bash">git clone https://github.com/marcelm/cutadapt
cd cutadapt
sudo python setup.py install
git clone https://github.com/FelixKrueger/TrimGalore
</code></pre>
<h2 id="deepvariant">DeepVariant</h2>
<p>It is a deep neural network to call genetic variants from next-generation DNA sequencing data, https://github.com/google/deepvariant.</p>
<h2 id="easyqc">EasyQC</h2>
<p>http://www.niot.res.in/EasyQC/</p>
<h2 id="exomiser">Exomiser</h2>
<pre><code class="language-bash">git clone https://github.com/exomiser/Exomiser
cd Exomiser
mvn package
</code></pre>
<p>See also https://github.com/exomiser/exomiser-demo.</p>
<h2 id="fastq-splitter">fastq-splitter</h2>
<p>The scripts divides a large FASTQ file into a set of smaller equally sized files, http://kirill-kryukov.com/study/tools/fastq-splitter/.</p>
<h2 id="fastx_toolkit-rsem">fastx_toolkit, RSEM</h2>
<p>It is also available from https://github.com/agordon/fastx_toolkit along with https://github.com/agordon/libgtextutils, and do away with the notorious automake-1.14 problem associated with sources at http://hannonlab.cshl.edu/fastx_toolkit/download.html.</p>
<p>However, line 105 of <code>src/fasta_formatter/fasta_formatter.cpp</code> requires <code>usage()</code> followed by <code>exit(0);</code> as suggested in the <code>issue</code> section. More oever, usage() is a void function so its own <code>exit(0)</code> is unnecessary.</p>
<p>The GitHub pages for RSEM are https://github.com/deweylab/RSEM and https://deweylab.github.io/RSEM/. It is also recommended that the Bioconductor package EBSeq be installed.</p>
<h2 id="freebayes">freebayes</h2>
<p>Try</p>
<pre><code class="language-bash">git clone --recursive https://github.com/ekg/freebayes
make
sudo make install
</code></pre>
<h2 id="gatk">GATK</h2>
<p>The source is available from https://github.com/broadinstitute/gatk/ but it is more convenient to use https://github.com/broadinstitute/gatk/releases/.</p>
<pre><code class="language-bash">ln -s `pwd`/gatk $HOME/bin/gatk
gatk --help
gatk --list
</code></pre>
<h2 id="hisat2-sambamba-picard-tools-stringtie">hisat2, sambamba, picard-tools, StringTie</h2>
<p>Except StringTie, this is overlapped with <code>apt install</code> above,</p>
<pre><code class="language-bash">brew tap brewsci/bio
brew tap brewsci/science
brew install hisat2
hisat2-build
brew install sambamba
brew install picard-tools
brew install stringtie
</code></pre>
<p>It could be useful with ``brew reinstall```. See</p>
<blockquote>
<p>Raghavachari N, Garcia-Reyero N (eds.) (2018), Gene Expression Analysis-Methods and Protocols, https://www.springer.com/us/book/9781493978335, Chapter 15, Springer.</p>
</blockquote>
<p>Nevertheless it may be slower, e.g., tophat, compared to <code>sudo apt install</code>.</p>
<h2 id="igv">IGV</h2>
<p>The download can be seeded from http://data.broadinstitute.org/igv, e.g., http://data.broadinstitute.org/igv/projects/downloads/2.4/IGV_2.4.10.zip.</p>
<p>Again the source code is from GitHub, https://github.com/igvteam/igv/. For developers, <a href="https://github.com/igvteam/igv.js">igv.js</a> is very appealing.</p>
<h2 id="jannovar">Jannovar</h2>
<p>From the GitHub repository, it is seen to use <code>project object model</code> (POM), an XML representation of a Maven project held in a file named <code>pom.xml</code>. We therefore install <code>maven</code> first,</p>
<pre><code class="language-bash">sudo apt install maven
</code></pre>
<p>The installation then proceeds as follows,</p>
<pre><code class="language-bash">git clone https://github.com/charite/jannovar
cd jannovar
mvn package
</code></pre>
<p>Other tasks such as compile, test, etc. are also possible.</p>
<p>It is handy to use symbolic link, i.e.,</p>
<pre><code class="language-bash">ln -s /home/jhz22/D/genetics/jannovar/jannovar-cli/target/jannovar-cli-0.24.jar $HOME/bin/Jannovar.jar
java -jar $HOME/bin/Jannovar.jar db-list
java -jar $HOME/bin/Jannovar.jar download -d hg19/refseq
</code></pre>
<p>We may need to set memory size, e.g., </p>
<pre><code class="language-bash">java -Xms2G -Xmx4G -jar $HOME/bin/Jannovar.jar
</code></pre>
<h1 id="melissa">Melissa</h1>
<p>https://github.com/andreaskapou/Melissa</p>
<blockquote>
<p>MEthyLation Inference for Single cell Analysis (Melissa), is a Bayesian hierarchical method to quantify spatially-varying methylation profiles across genomic regions from single-cell bisulfite sequencing data (scBS-seq). Melissa clusters individual cells based on local methylation patterns, enabling the discovery of epigenetic diversities and commonalities among individual cells. The clustering also acts as an effective regularisation method for imputation of methylation on unassayed CpG sites, enabling transfer of information between individual cells.</p>
</blockquote>
<p>Kapourani C-A, Sanguinetti G (2019). Melissa: Bayesian clustering and imputation of single-cell methylomes, <em>Genome Biology</em> 20:61, https://doi.org/10.1186/s13059-019-1665-8</p>
<h1 id="monster">MONSTER</h1>
<p>http://galton.uchicago.edu/~mcpeek/software/MONSTER/
http://galton.uchicago.edu/~mcpeek/software/MONSTER/MONSTER_v1.3.tar.gz</p>
<p>Jiang D, McPeek MS (2014). Robust Rare Variant Association Testing for Quantitative Traits in Samples with Related Individuals. <em>Genetic Epidemiology</em> 38(1):10-20</p>
<h1 id="pindel">pindel</h1>
<p>The software can be obtained from https://github.com/genome/pindel.</p>
<p>After htslib is installed, the canonical instruction is to issue</p>
<pre><code class="language-bash">git clone https://github.com/samtools/htslib
cd htslib
make
sudo make install
cd -
git clone https://github.com/genome/pindel
cd pindel
./INSTALL ../htslib
</code></pre>
<p>It is 'standard' to have complaints about pindel.cpp, bddate.cpp and genotyping.cpp,
for <code>abs()</code> rather than <code>fabs()</code> from the header file <code>cmath</code> have been used. The
issue goes away when <code>abs</code> is replaced with <code>fabs</code> and in the case of bddata.cpp,
it is also necessary to invoke the header, i.e.,</p>
<pre><code class="language-cpp">#include &lt;cmath&gt;
</code></pre>
<h2 id="rtg-tools">rtg-tools</h2>
<p>It is available from https://www.realtimegenomics.com/products/rtg-tools and GitHub,</p>
<pre><code class="language-bash">git clone https://github.com/RealTimeGenomics/rtg-tools.git
ant
dir dist
</code></pre>
<h2 id="sambamba">sambamba</h2>
<p>While the source contains ldc2, it is readily available with Ubuntu archive nevertheless failed to compile, 
so we proceed with instructions at the GitHub, e.g.,</p>
<pre><code class="language-bash">export PATH=$HOME/ldc2-1.10.0-linux-x86_64/bin:$PATH
export LIBRARY_PATH=$HOME/ldc2-1.10.0-linux-x86_64/lib
</code></pre>
<p>for version 1.10.0.</p>
<h2 id="samtools">samtools</h2>
<p>To build from source, we do these,</p>
<pre><code class="language-bash">git clone https://github.com/samtools/htslib
cd htslib
make
cd -
git clone https://github.com/samtools/samtools
cd samtools
autoheader            # Build config.h.in (this may generate a warning about
                      # AC_CONFIG_SUBDIRS - please ignore it).
autoconf -Wno-syntax  # Generate the configure script
./configure           # Needed for choosing optional functionality
make
make install
</code></pre>
<p>Note bgzip and tabix are distributed with htslib. It is relatively easier to install from release,</p>
<pre><code class="language-bash">wget https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2
tar xjf samtools-1.9.tar.gz
cd samtools-1.9
./configure --prefix=/scratch/jhz22
make
make install
cd -
cd htslib-1.9
./configure --prefix=/scratch/jhz22
make
make install
</code></pre>
<p>where we install tabix as well.</p>
<h2 id="snpeff-snpsift-clineff">SnpEff, SnpSift, clinEff</h2>
<p>It is straightforward with the compiled version from sourceforge, which also includes clinEff.</p>
<pre><code class="language-bash">wget http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip
unzip snpEff_latest_core
cd snpEff
java -jar snpEff.jar databases
java -jar snpEff.jar download GRCh38.76
wget http://sourceforge.net/projects/snpeff/files/databases/test_cases.tgz
tar fvxz test_cases.tgz
</code></pre>
<p>lists all the databases and download a particular one. Later, the test files are also downloaded and extracted.</p>
<p>The following steps compile from source instead.</p>
<pre><code class="language-bash">git clone https://github.com/pcingola/SnpEff.git
cd SnpEff
mvn package
mvn install
cd lib
# Antlr
mvn install:install-file \
    -Dfile=antlr-4.5.1-complete.jar \
    -DgroupId=org.antlr \
    -DartifactId=antlr \
    -Dversion=4.5.1 \
    -Dpackaging=jar

# BioJava core
mvn install:install-file \
    -Dfile=biojava3-core-3.0.7.jar \
    -DgroupId=org.biojava \
    -DartifactId=biojava3-core \
    -Dversion=3.0.7 \
    -Dpackaging=jar

# BioJava structure
mvn install:install-file \
    -Dfile=biojava3-structure-3.0.7.jar \
    -DgroupId=org.biojava \
    -DartifactId=biojava3-structure \
    -Dversion=3.0.7 \
    -Dpackaging=jar

cd -

# SnpSift
git clone https://github.com/pcingola/SnpSift.git
cd SnpSift
mvn package
mvn install
</code></pre>
<p>which gives <code>target/SnpEff-4.3.jar</code> and <code>target/SnpSift-4.3.jar</code>, respectively.</p>
<p>Note that <code>antlr4</code> is from GitHub, https://github.com/antlr/antlr4. See also https://github.com/sanger-pathogens/SnpEffWrapper.</p>
<h2 id="subread">subread</h2>
<p>It is available from http://subread.sourceforge.net/.</p>
<h2 id="svict">SViCT</h2>
<p>Short for Structural Variant detrction in Circulating Tumor DNA and is available from</p>
<p>https://github.com/vpc-ccg/svict</p>
<h2 id="tagdust">tagdust</h2>
<p>http://sourceforge.net/projects/tagdust/</p>
<h2 id="trinity">Trinity</h2>
<p>RNA-Seq De novo Assembly Using Trinity, https://github.com/trinityrnaseq/trinityrnaseq/wiki.</p>
<h2 id="varscan">VarScan</h2>
<p>Hosted at https://github.com/dkoboldt/varscan, the .jar files are ready to use with</p>
<pre><code class="language-bash">git clone https://github.com/dkoboldt/varscan
</code></pre>
<p>or from the repository releases.</p>
<p>See http://varscan.sourceforge.net/ for further information.</p>
<h2 id="vcftools">vcftools</h2>
<p>Assuming that we use zlib 1.2.8 from module zlib/1.2.8, we can do the following,</p>
<pre><code class="language-bash">wget https://github.com/vcftools/vcftools/releases/download/v0.1.16/vcftools-0.1.16.tar.gz
tar xvfz vcftools-0.1.16.tar.gz
module load zlib/1.2.8
./configure --prefix=/scratch/jhz22 ZLIB_CFLAGS=&quot;-I/usr/local/Cluster-Apps/zlib/1.2.8/include&quot; ZLIB_LIBS=&quot;-L/usr/local/Cluster-Apps/zlib/1.2.8/lib -lz&quot;
make
make install
</code></pre>
<p>To use vcf-concat, it is necessary to set the PERL5LIB environment variables, e.g.,</p>
<pre><code class="language-bash">export PERL5LIB=/scratch/jhz22/share/perl5
</code></pre>
<h2 id="wasp">WASP</h2>
<p>Allele-specific pipeline for unbiased read mapping and molecular QTL discovery, https://github.com/bmvdgeijn/WASP/.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../SL/" class="btn btn-neutral float-right" title="SL">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../CRISPR/" class="btn btn-neutral" title="CRISPR"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../CRISPR/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../SL/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
